### ISSUE TITLE
WebXR Raw Camera Access API

### ISSUE THREAD

**Issue by bialpio** [mozilla/standards-positions#667]
## Request for Mozilla Position on an Emerging Web Specification

* Specification Title: WebXR Raw Camera Access Module
* Specification or proposal URL: https://immersive-web.github.io/raw-camera-access/
* Caniuse.com URL (optional): https://caniuse.com/webxr (the Raw Camera Access Module extends WebXR Device API)
* Bugzilla URL (optional): https://bugzilla.mozilla.org/show_bug.cgi?id=1419190 (the Raw Camera Access Module extends WebXR Device API)
* Mozillians who can provide input (optional): 

### Other information

The WebXR Raw Camera Access Module extends the capabilities of the core WebXR Device API by allowing the sites to request `"camera-access"` feature when creating XR sessions. The feature allows the sites to obtain access to camera pixels (via an integration with WebGL, exposing the pixels as `WebGLTexture`). This capability has been requested for a long time by developers.

**Comment by ekr** [comment]
Perhaps a stupid question, but how is this different from existing camera access APIs such as `getUserMedia()`

**Comment by bialpio** [comment]
It allows the sites to obtain camera images that are synchronized with WebXR's `XRPose`s. If a site were to obtain camera images in some other manner (like `getUserMedia()`), it would be unable to correlate them with spatial data that it can get by integrating with WebXR.

**Comment by ekr** [comment]
Does it otherwise provide the same data as `getUserMedia()`?

**Comment by bialpio** [comment]
No, there are some differences. The big one is that the camera image that is presented to the site is going to be cropped compared to the image available via `getDisplayMedia()`. The spec mandates that only the part of the image is [aligned](https://immersive-web.github.io/raw-camera-access/#aligned) to the `XRView` will be accessible to the site, and in smartphone AR case, that cropped camera image is also displayed to the user. We also expose the projection matrix of the camera (since it's the same as the one on `XRView` due to the alignment requirement). The other difference is the way that the camera image is exposed - we surface the image as `WebGLTexture`, which the site can then use for rendering - this should be a more direct approach than obtaining `ImageBitmap` (via `ImageCapture` constructed from `MediaStreamTrack`) & uploading it to a `WebGLTexture`.

**Comment by zcorpan** [comment]
Marking this "position: defer" because Mozilla is not working on WebXR currently.

You could consider asking Wolvic: https://github.com/Igalia/wolvic/issues
